src/Groups/GAPGroups.jl

```diff
@@ -748,9 +947,10 @@ function sylow_subgroup(G::GAPGroup, p::Int64)
 end
 
 """
-    hall_subgroup(G::Group, P::Array{Int64})
+    hall_subgroup(G::Group, P::Vector{Int64})
 
-Return a Hall `P`-subgroup of `G`. It works only if `G` is solvable.
+Return a Hall `P`-subgroup of `G`.
+An exception is thrown if `G` is not solvable.
```

Comment:
This indeed describes what the code does, but... why actually does it? In GAP, I think we support non-solvable groups, too, at least if they have suitable subgroups? Perhaps we don't want to support this here -- fine, but then we should perhaps still elaborate *why*, i.e. why we throw an exception here?

Reply:
@fingolfin The GAP function `HallSubgroup` works also for (finite) nonsolvable groups, and returns either a Hall P-subgroup (if it is unique up to conjugacy), or a list of Hall P-subgroups (representatives of G-conjugacy classes of these subgroups if there is more than one class), or `fail` (if no Hall P-subgroup exists).

For Oscar, we could introduce a function `representatives_hall_subgroups`, which always returns a vector of representatives of G-conjugacy classes of Hall P-subgroups.